src/main/kotlin/com/terraformation/backend/support/atlassian/AtlassianHttpClient.kt

```diff
@@ -0,0 +1,110 @@
+package com.terraformation.backend.support.atlassian
+
+import com.terraformation.backend.config.TerrawareServerConfig
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.support.atlassian.model.ServiceDeskModel
+import com.terraformation.backend.support.atlassian.model.ServiceRequestFieldsModel
+import com.terraformation.backend.support.atlassian.model.ServiceRequestTypeModel
+import com.terraformation.backend.support.atlassian.resource.AtlassianResource
+import com.terraformation.backend.support.atlassian.resource.CreateServiceDeskRequest
+import com.terraformation.backend.support.atlassian.resource.DeleteIssue
+import com.terraformation.backend.support.atlassian.resource.ListServiceDesks
+import com.terraformation.backend.support.atlassian.resource.ListServiceRequestTypes
+import com.terraformation.backend.support.atlassian.resource.PostServiceDeskRequestResponse
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.java.Java
+import io.ktor.client.plugins.auth.*
+import io.ktor.client.plugins.auth.providers.*
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.plugins.defaultRequest
+import io.ktor.client.request.accept
+import io.ktor.client.request.request
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import io.ktor.serialization.jackson.JacksonConverter
+import jakarta.inject.Named
+import kotlinx.coroutines.runBlocking
+
+/** Submits API requests to interact with Atlassian services. */
+@Named
+class AtlassianHttpClient(private val config: TerrawareServerConfig) {
+  private val httpClient: HttpClient by lazy { createHttpClient() }
+  private val serviceDesk: ServiceDeskModel? by lazy { findServiceDesk() }
+
+  val requestTypes: Set<ServiceRequestTypeModel> by lazy { getServiceRequestTypes() }
```

Comment:
Could this be `Map<Int, ServiceRequestTypeModel>` or even `Set<Int>` instead? Or will it eventually be used for more than just validating that the request type ID exists?

Reply:
So I am imagining there will be a GET /support that will return the list of support request types configured on the Jira, and PUT /support that will create an issue. Routes are subject to changed.